core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -568,8 +574,9 @@ impl<Block: BlockT> Inner<Block> {
 			set_id: self.local_view.set_id,
 			commit_finalized_height: self.local_view.last_commit.unwrap_or(Zero::zero()),
 		};
-
-		let peers = self.peers.inner.keys().cloned().collect();
+		let peers = self.peers.inner.iter().filter(|(_, peer_info)| (
+			peer_info.view.is_valid_update(packet.set_id, packet.round, &packet.commit_finalized_height)
```

Comment:
this is a bit of a hack I think. can we identify the message we send which is not a valid update? our local view should never update in a way which is invalid.